### PR TITLE
(feat) Redirect users to the page they were on before being logged out

### LINKED ISSUE #924: Redirect users to the page they were on before being logged out
## **Describe the the feature request**
Our current application lacks a mechanism to preserve the user's original URL upon logout, either by manual action or automatic session expiration. This functionality is pivotal for maintaining a consistent user experience. Upon reauthentication, users should seamlessly resume their workflow by being redirected to the exact URL they were on before logging out, ensuring uninterrupted continuity

## Tasks

- [x] **Capture Initial URL:**
    - Implement a function/component to capture the current URL and store it in the browser's session storage or a temporary state when the user logs out or is automatically logged out.
  
- [x] **Redirect After Login:**
    - Modify the login logic to check if there's a stored initial URL in session storage or state.
    - If a stored URL is found, redirect the user to that URL after a successful login.
    - If no stored URL is found, redirect the user to the default landing page.
  
- [x] **Edge Cases Handling:**
    - Handle edge cases, such as the initial URL being invalid or expired, ensuring graceful degradation of the user experience.
  
- [ ] **Testing:**
    - Write tests to validate the behavior of the URL redirection feature in different scenarios, including successful login, expired 
       initial URL, and invalid initial URL.

- [x] **Documentation:**
     - Update the project documentation to include information about this new feature, explaining its functionality and usage.

### PR DESCRIPTION
**Pull request by coderatomy**
## Summary
Save the `initalUrl` in the `sessionStorage` when logging out and removes it immediately after logging in
Closes #924 

## Changes
- Save the `initialUrl` in sessionStorage 

## Screencasts
[Screencast from 2023-10-20 08-30-06.webm](https://github.com/unstructuredstudio/zubhub/assets/141822315/7ae8abbb-038b-4374-b2b5-01291d642ad9)

### CODE REVIEW

File: zubhub_frontend/zubhub/src/store/actions/authActions.js [hunk 1]
```diff
@@ -37,7 +37,15 @@ export const login = args => {
           payload: { token: res.key },
         });
       })
-      .then(() => args.history.push('/'));
+      .then(() => {
+        const initialUrl = sessionStorage.getItem('initialUrl');
```
Comment (coderatomy) [inline zubhub_frontend/zubhub/src/store/actions/authActions.js:41]: getting the `initailUrl` from the sessionStorage

File: zubhub_frontend/zubhub/src/api/api.js [hunk 2]
```diff
@@ -124,6 +124,9 @@ class API {
    * @todo - describe method's signature
    */
   logout = token => {
+    const initialUrl = window.location.href
+    sessionStorage.setItem('initialUrl', initialUrl)
```
Comment (coderatomy) [inline zubhub_frontend/zubhub/src/api/api.js:128]: saving the `initialUrl` in sessionStorage when logging out

File: zubhub_frontend/zubhub/src/App.js [hunk 3]
```diff
@@ -1,7 +1,7 @@
 import React, { useContext } from 'react';
 import { withTranslation } from 'react-i18next';
 import { BrowserRouter as Router, Route, Redirect, Switch } from 'react-router-dom';
-import CreateActivity from './views/create_activity/create_activity';
+import CreateActivity from './views/create_activity/CreateActivity';
```
Comment (tuxology) [inline zubhub_frontend/zubhub/src/App.js:4]: 😏

**Review by tuxology — APPROVED** [review]
LGTM 🌈

File: zubhub_frontend/zubhub/src/App.js (hunk 3 above)
Comment (coderatomy) [reply zubhub_frontend/zubhub/src/App.js:4]: Won't conflict. I merged the regression commit into this since it was blocking.

**Review by tuxology — APPROVED** [review]
